workers/common/src/main/kotlin/common/env/NetRcManager.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.workers.common.env
+
+import java.net.URI
+
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+import org.eclipse.apoapsis.ortserver.model.CredentialsType
+import org.eclipse.apoapsis.ortserver.model.InfrastructureService
+import org.eclipse.apoapsis.ortserver.utils.logging.runBlocking
+import org.eclipse.apoapsis.ortserver.workers.common.context.AuthenticationEvent
+import org.eclipse.apoapsis.ortserver.workers.common.context.AuthenticationListener
+import org.eclipse.apoapsis.ortserver.workers.common.context.WorkerContext
+import org.eclipse.apoapsis.ortserver.workers.common.env.definition.EnvironmentServiceDefinition
+
+import org.slf4j.LoggerFactory
+
+/**
+ * An internal helper class to dynamically update the content of the `.netrc` file when there is change in the
+ * authentication state.
+ *
+ * An instance of this class is created when setting up the environment for executing a worker. It is initialized with
+ * the set of infrastructure services currently available. It is also registered at the central authenticator used by
+ * ORT Server. Its task is to figure out, based on authentication events, which infrastructure services are currently
+ * referenced. These are then written into the `.netrc` file, if they have the corresponding credentials type. This
+ * should prevent conflicts with multiple services for the same host.
+ */
+internal class NetRcManager(
+    /** The current [WorkerContext]. */
+    private val context: WorkerContext,
+
+    /** The currently known set of infrastructure services. */
+    services: Collection<InfrastructureService>
+) : AuthenticationListener {
+    companion object {
+        private val logger = LoggerFactory.getLogger(NetRcManager::class.java)
+
+        /**
+         * Create a new instance of [NetRcManager] and initialize it with the given [context] and [services].
+         */
+        fun create(context: WorkerContext, services: Collection<InfrastructureService>): NetRcManager =
+            NetRcManager(context, services)
```

Comment:
Why do we need a factory method that's exactly the same as the constructor of the class?

Reply:
The factory function simplifies testing the integration of `NetRcManager` with the environment service (see later commit). Otherwise, constructor mocks would have to be used, which are always hard to deal with.

Reply:
Ideally, the design of the feature would not require that much mocking in the first place.